program-runtime/src/timings.rs

```diff
@@ -34,34 +36,59 @@ impl ProgramTiming {
     }
 }
 
-#[derive(Default, Debug)]
+#[derive(Debug)]
+pub enum ExecuteTimingType {
+    CheckUs,
+    LoadUs,
+    ExecuteUs,
+    StoreUs,
+    UpdateStakesCacheUs,
+    NumExecuteBatches,
+    CollectLogsUs,
+}
+
+type Metrics = [u64; 7];
+
+#[derive(Debug, Default)]
 pub struct ExecuteTimings {
-    pub check_us: u64,
-    pub load_us: u64,
-    pub execute_us: u64,
-    pub store_us: u64,
-    pub update_stakes_cache_us: u64,
+    pub metrics: Metrics,
     pub total_batches_len: usize,
-    pub num_execute_batches: u64,
-    pub collect_logs_us: u64,
     pub details: ExecuteDetailsTimings,
     pub execute_accessories: ExecuteAccessoryTimings,
 }
 
 impl ExecuteTimings {
     pub fn accumulate(&mut self, other: &ExecuteTimings) {
-        saturating_add_assign!(self.check_us, other.check_us);
-        saturating_add_assign!(self.load_us, other.load_us);
-        saturating_add_assign!(self.execute_us, other.execute_us);
-        saturating_add_assign!(self.store_us, other.store_us);
-        saturating_add_assign!(self.update_stakes_cache_us, other.update_stakes_cache_us);
-        saturating_add_assign!(self.total_batches_len, other.total_batches_len);
-        saturating_add_assign!(self.num_execute_batches, other.num_execute_batches);
-        saturating_add_assign!(self.collect_logs_us, other.collect_logs_us);
+        if self.metrics.len() == other.metrics.len() {
+            for (i, timing) in self.metrics.iter_mut().enumerate() {
```

Comment:
Should be able to `.zip(other.metrics.iter()` instead of `enumerate()`.  With `zip()`, the length check above are unnecessary (pretty sure type checker ensures it anyway).

Reply:
wow this is convenient, thanks for the suggestion